src/ynViewer.sol

```diff
@@ -1,20 +1,27 @@
 // SPDX-License-Identifier: BSD 3-Clause License
 pragma solidity ^0.8.24;
 
-import {IynETH} from "src/interfaces/IynETH.sol";
-import {IynViewer} from "src/interfaces/IynViewer.sol";
-import {IStakingNodesManager,IStakingNode} from "src/interfaces/IStakingNodesManager.sol";
+import {IDelegationManager, IStrategy} from "@eigenlayer/src/contracts/interfaces/IDelegationManager.sol";
+import {IEigenPodManager} from "@eigenlayer/src/contracts/interfaces/IEigenPodManager.sol";
+
+import {IStakingNode, IStakingNodesManager, IynViewer} from "./interfaces/IynViewer.sol";
+
+interface IynETH {
+    function totalSupply() external view returns (uint256);
+    function totalAssets() external view returns (uint256);
+}
 
 contract ynViewer is IynViewer {
-    IynETH public ynETH;
-    IStakingNodesManager public stakingNodesManager;
+
+    IynETH public immutable ynETH;
+    IStakingNodesManager public immutable stakingNodesManager;
 
     /// @notice Initializes a new ynViewer contract.
     /// @param _ynETH The address of the ynETH contract.
     /// @param _stakingNodesManager The address of the StakingNodesManager contract.
-    constructor(IynETH _ynETH, IStakingNodesManager _stakingNodesManager) {
-        ynETH = _ynETH;
-        stakingNodesManager = _stakingNodesManager;
+    constructor(address _ynETH, address _stakingNodesManager) {
```

Comment:
let's use 

    constructor() {
       _disableInitializers();
    }


and initialize these values separately with an initializer